web/src/engine/websites/Kakaopage.ts

```diff
@@ -0,0 +1,220 @@
+            node: {
+                single: {
+                    productId: number,
+                    title: string
+                }
+            }
+        }[],
+        pageInfo: {
+            hasNextPage: boolean,
+            endCursor: string
+        }
+    }
+}
+
+type APiPages = {
+    viewerInfo: {
+        viewerData: {
+            imageDownloadData: {
+                files: {
+                    secureUrl: string
+                }[]
+            }
+        }
+    }
+}
+
+@Common.MangasNotSupported()
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+    public constructor() {
+        super('kakaopage', `Page Kakao (카카오페이지)`, 'https://page.kakao.com', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Korean, Tags.Source.Official);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return new RegExp(`^${this.URI.origin}/content/[\\d]+$`).test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const id = parseInt(new URL(url).pathname.match(/content\/(\d+)/)[1]);
+        const gql = `
+            query contentHomeOverview($seriesId: Long!) {
+              contentHomeOverview(seriesId: $seriesId) {
+                content {
+                  ...SeriesFragment
+                }
+              }
+            }
+            fragment SeriesFragment on Series {
+              id
+              seriesId
+              title
+            }
+        `;
+        const vars = { seriesId: id };
+        const request = new FetchRequest(new URL('/graphql', this.URI).href, {
+            headers: {
+                Referer: this.URI.href,
+                Origin: this.URI.href
+            }
+        });
+        const operationName = 'contentHomeOverview';
+        const { contentHomeOverview } = await FetchGraphQL<ApiClipBoard>(request, operationName, gql, JSON.stringify(vars));
+        return new Manga(this, provider, String(contentHomeOverview.content.seriesId), contentHomeOverview.content.title.trim());
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        let nextCursor = '';
+        const chapterList = [];
+        for (let run = true; run;) {
+            const data = await this.getChaptersFromPage(manga, nextCursor);
+            const chapters = data.contentHomeProductList.edges.map(chapter => new Chapter(this, manga, chapter.node.single.productId.toString(), chapter.node.single.title.replace(manga.Title, '').trim()));
+            nextCursor = data.contentHomeProductList.pageInfo.hasNextPage ? data.contentHomeProductList.pageInfo.endCursor : null;
+            chapterList.push(...chapters);
+            run = nextCursor != null;
+        }
+        return chapterList;
+    }
+
+    async getChaptersFromPage(manga: Manga, nextCursor: string): Promise<APIChapters> {
```

Comment:
Usually the `get*FromPage` methods are already returning the target type e.g., `Promise<Manga[]>` or `Promise<Chapter[]>`

Reply:
Yup but we need the cursor in the loop here, which is arbitrary and not a counter iirc.